### PR TITLE
Feat: add avatar equip slots animations for interactions

### PR DESCRIPTION
**Pull request by davidejensen**
## What does this PR change?

Adds avatar equip slots animations, cannot be tested

https://user-images.githubusercontent.com/4254522/233325742-d3c92571-662d-45bc-9403-262ad54a9d90.mov

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 1dfccf6</samp>

Added animations and a new property to the backpack editor HUD. The animations use `DOTween` to enhance the user feedback when interacting with the avatar slots. The `nftContainer` property allows the NFT image to be referenced in the code.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotComponentView.cs [hunk 1]
```diff
@@ -107,7 +113,21 @@ public override void OnLoseFocus()
...
+                selectedImage.transform.DOScale(1, ANIMATION_TIME).SetEase(Ease.OutBack);
+            }
+            else
+            {
+                selectedImage.transform.DOScale(0, ANIMATION_TIME).SetEase(Ease.OutBack).OnComplete(() =>
+                {
+                    selectedImage.enabled = false;
+                    selectedImage.transform.localScale = new Vector3(1,1,1);
+                });
+            }
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotComponentView.cs:130]: Can we extract all the animations logic into separated methods? In this way we make the code cleaner and understandable. Wdyt?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotComponentView.cs [hunk 2]
```diff
@@ -96,6 +100,8 @@ public override void OnFocus()
         {
             focusedImage.enabled = true;
             tooltipContainer.SetActive(true);
+            focusedImage.transform.localScale = new Vector3(0, 0, 0);
+            focusedImage.transform.DOScale(1, ANIMATION_TIME).SetEase(Ease.OutBack);
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotComponentView.cs:104]: Can we extract all the animations logic into separated methods? In this way we make the code cleaner and understandable. Wdyt?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotComponentView.cs [hunk 3]
```diff
@@ -51,9 +55,9 @@ public void SetIsHidden(bool isHidden, string hiddenBy)
             model.isHidden = isHidden;
             model.hiddenBy = hiddenBy;
             hiddenSlot.SetActive(isHidden);
-
             if (isHidden)
             {
+                nftContainer.DOShakePosition(SHAKE_ANIMATION_TIME, 4);
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotComponentView.cs:60]: Can we extract all the animations logic into separated methods? In this way we make the code cleaner and understandable. Wdyt?

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]
Can we extract all the animations logic into separated methods? In this way we make the code cleaner and understandable. Wdyt?

**Review by sandrade-dcl — APPROVED** [review]

**Review by lorux0 — APPROVED** [review]

**Pull request merged**